Build failure: zeal 0.8.1 (darwin)

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix build -v github:nixos/nixpkgs/82a41ced88dfb290e9e873e859fa3c4f620d19b1#zeal
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/325815365

### Relevant log output

```console
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/gzavc4qkxq12b7zncqqscbn2mxg9wv2v-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/doc/Zeal -DCMAKE_INSTALL_INFODIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib -DCMAKE_STRIP=/nix/store/s7qlr26bmc6n4r607scz8iiwcg6yg4ic-clang-wrapper-21.1.8/bin/strip -DCMAKE_RANLIB=/nix/store/s7qlr26bmc6n4r607scz8iiwcg6yg4ic-clang-wrapper-21.1.8/bin/ranlib -DCMAKE_AR=/nix/store/s7qlr26bmc6n4r607scz8iiwcg6yg4ic-clang-wrapper-21.1.8/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1 -DZEAL_RELEASE_BUILD:BOOL=TRUE -DKDE_INSTALL_EXECROOTDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1 -DKDE_INSTALL_BINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/bin -DKDE_INSTALL_SBINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/sbin -DKDE_INSTALL_LIBDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib -DKDE_INSTALL_LIBEXECDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/libexec -DKDE_INSTALL_CMAKEPACKAGEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib/cmake -DKDE_INSTALL_INCLUDEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/include -DKDE_INSTALL_LOCALSTATEDIR=/var -DKDE_INSTALL_DATAROOTDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share -DKDE_INSTALL_DATADIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share -DKDE_INSTALL_DOCBUNDLEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/doc/HTML -DKDE_INSTALL_KCFGDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/config.kcfg -DKDE_INSTALL_KCONFUPDATEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/kconf_update -DKDE_INSTALL_KSERVICES5DIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/kservices5 -DKDE_INSTALL_KSERVICETYPES5DIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/kservicetypes5 -DKDE_INSTALL_KXMLGUI5DIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/kxmlgui5 -DKDE_INSTALL_KNOTIFY5RCDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/knotifications5 -DKDE_INSTALL_ICONDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/icons -DKDE_INSTALL_LOCALEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/locale -DKDE_INSTALL_SOUNDDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/sounds -DKDE_INSTALL_TEMPLATEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/templates -DKDE_INSTALL_WALLPAPERDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/wallpapers -DKDE_INSTALL_APPDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/applications -DKDE_INSTALL_DESKTOPDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/desktop-directories -DKDE_INSTALL_MIMEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/mime/packages -DKDE_INSTALL_METAINFODIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/metainfo -DKDE_INSTALL_MANDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/man -DKDE_INSTALL_INFODIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/info -DKDE_INSTALL_DBUSDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/dbus-1 -DKDE_INSTALL_DBUSINTERFACEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/dbus-1/interfaces -DKDE_INSTALL_DBUSSERVICEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/dbus-1/services -DKDE_INSTALL_DBUSSYSTEMSERVICEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/share/dbus-1/system-services -DKDE_INSTALL_SYSCONFDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/etc -DKDE_INSTALL_CONFDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/etc/xdg -DKDE_INSTALL_AUTOSTARTDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/etc/xdg/autostart -DKDE_INSTALL_BUNDLEDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/Applications/KDE -DKDE_INSTALL_QTPLUGINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib/qt-6/plugins -DKDE_INSTALL_PLUGINDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib/qt-6/plugins -DKDE_INSTALL_QMLDIR=/nix/store/61zvazvb9kjidyi948c0257jgjp71vv1-zeal-0.8.1/lib/qt-6/qml -DECM_DIR=/nix/store/p3fsysv6qd1md3775vckkis75ln1kwlv-extra-cmake-modules-5.116.0/share/ECM/cmake
-- The CXX compiler identification is Clang 21.1.8
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/s7qlr26bmc6n4r607scz8iiwcg6yg4ic-clang-wrapper-21.1.8/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
Detected Qt version: 6.10.2
Release build: 0.8.1
Calculated Zeal version: 0.8.1
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/q2dccg26bm7bn6ia1q30qkl5jck7wwgb-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/4jq007kmqkhfby8d5zcm57rcvnjd41p2-vulkan-headers-1.4.341.0/include
-- Found Cups: /nix/store/zrgjcm76i0k3ipzaksryp7czgh6v07s6-cups-2.4.16-lib/lib/libcups.dylib (found version "2.4.16")
-- Found SQLite3: /nix/store/w4g9wg8n4f85cialnsdwriwlkjdff9b5-sqlite-3.51.2/lib/libsqlite3.dylib (found version "3.51.2")
-- Project executable name: Zeal
Detected locale "C" with character encoding "US-ASCII", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
-- Configuring done (6.8s)
-- Generating done (0.3s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    ECM_DIR
    KDE_INSTALL_APPDIR
    KDE_INSTALL_AUTOSTARTDIR
    KDE_INSTALL_BINDIR
    KDE_INSTALL_BUNDLEDIR
    KDE_INSTALL_CMAKEPACKAGEDIR
    KDE_INSTALL_CONFDIR
    KDE_INSTALL_DATADIR
    KDE_INSTALL_DATAROOTDIR
    KDE_INSTALL_DBUSDIR
    KDE_INSTALL_DBUSINTERFACEDIR
    KDE_INSTALL_DBUSSERVICEDIR
    KDE_INSTALL_DBUSSYSTEMSERVICEDIR
    KDE_INSTALL_DESKTOPDIR
    KDE_INSTALL_DOCBUNDLEDIR
    KDE_INSTALL_EXECROOTDIR
    KDE_INSTALL_ICONDIR
    KDE_INSTALL_INCLUDEDIR
    KDE_INSTALL_INFODIR
    KDE_INSTALL_KCFGDIR
    KDE_INSTALL_KCONFUPDATEDIR
    KDE_INSTALL_KNOTIFY5RCDIR
    KDE_INSTALL_KSERVICES5DIR
    KDE_INSTALL_KSERVICETYPES5DIR
    KDE_INSTALL_KXMLGUI5DIR
    KDE_INSTALL_LIBDIR
    KDE_INSTALL_LIBEXECDIR
    KDE_INSTALL_LOCALEDIR
    KDE_INSTALL_LOCALSTATEDIR
    KDE_INSTALL_MANDIR
    KDE_INSTALL_METAINFODIR
    KDE_INSTALL_MIMEDIR
    KDE_INSTALL_PLUGINDIR
    KDE_INSTALL_QMLDIR
    KDE_INSTALL_QTPLUGINDIR
    KDE_INSTALL_SBINDIR
    KDE_INSTALL_SOUNDDIR
    KDE_INSTALL_SYSCONFDIR
    KDE_INSTALL_TEMPLATEDIR
    KDE_INSTALL_WALLPAPERDIR
    ZEAL_RELEASE_BUILD


-- Build files have been written to: /nix/var/nix/builds/nix-60139-4043517989/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/s0psayl7zvkvwdcqc8fy1sbv8rlf1yq8-bash-5.3p9/bin/bash
[  0%] Built target Sidebar_autogen_timestamp_deps
[  1%] Built target Browser_autogen_timestamp_deps
[  1%] Running qmlimportscanner for App
[  1%] Built target Util_autogen_timestamp_deps
[  1%] Built target Widgets_autogen_timestamp_deps
[  1%] Built target QxtGlobalShortcut_autogen_timestamp_deps
[  2%] Automatic MOC and UIC for target Widgets
[  4%] Automatic MOC and UIC for target Browser
[  4%] Automatic MOC and UIC for target Sidebar
[  7%] Automatic MOC and UIC for target Util
[  7%] Automatic MOC and UIC for target QxtGlobalShortcut
[  7%] Built target App_qmlimportscan
[  7%] Built target Util_autogen
[  8%] Building CXX object src/libs/util/CMakeFiles/Util.dir/Util_autogen/mocs_compilation.cpp.o
[  9%] Building CXX object src/libs/util/CMakeFiles/Util.dir/fuzzy.cpp.o
[ 10%] Building CXX object src/libs/util/CMakeFiles/Util.dir/plist.cpp.o
[ 12%] Building CXX object src/libs/util/CMakeFiles/Util.dir/humanizer.cpp.o
[ 13%] Building CXX object src/libs/util/CMakeFiles/Util.dir/sqlitedatabase.cpp.o
[ 13%] Built target QxtGlobalShortcut_autogen
[ 13%] Built target Sidebar_autogen
[ 13%] Built target Widgets_autogen
[ 14%] Building CXX object src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/QxtGlobalShortcut_autogen/mocs_compilation.cpp.o
[ 16%] Building CXX object src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/qxtglobalshortcut_mac.cpp.o
[ 16%] Building CXX object src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/qxtglobalshortcut.cpp.o
[ 16%] Built target Browser_autogen
[ 18%] Building CXX object src/libs/sidebar/CMakeFiles/Sidebar.dir/Sidebar_autogen/mocs_compilation.cpp.o
[ 19%] Building CXX object src/libs/sidebar/CMakeFiles/Sidebar.dir/container.cpp.o
[ 21%] Building CXX object src/libs/ui/widgets/CMakeFiles/Widgets.dir/Widgets_autogen/mocs_compilation.cpp.o
[ 21%] Building CXX object src/libs/ui/widgets/CMakeFiles/Widgets.dir/shortcutedit.cpp.o
[ 22%] Building CXX object src/libs/ui/widgets/CMakeFiles/Widgets.dir/layouthelper.cpp.o
[ 24%] Building CXX object src/libs/sidebar/CMakeFiles/Sidebar.dir/proxyview.cpp.o
[ 25%] Building CXX object src/libs/sidebar/CMakeFiles/Sidebar.dir/view.cpp.o
[ 26%] Building CXX object src/libs/ui/widgets/CMakeFiles/Widgets.dir/searchedit.cpp.o
[ 27%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/Browser_autogen/mocs_compilation.cpp.o
[ 28%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/searchtoolbar.cpp.o
[ 30%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/settings.cpp.o
[ 31%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/urlrequestinterceptor.cpp.o
[ 32%] Linking CXX static library libUtil.a
[ 33%] Built target Util
[ 34%] Building CXX object src/libs/ui/widgets/CMakeFiles/Widgets.dir/toolbarframe.cpp.o
[ 36%] Built target Registry_autogen_timestamp_deps
[ 36%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/webbridge.cpp.o
[ 37%] Automatic MOC and UIC for target Registry
[ 38%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/webcontrol.cpp.o
[ 39%] Building CXX object src/libs/sidebar/CMakeFiles/Sidebar.dir/viewprovider.cpp.o
[ 40%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/webpage.cpp.o
[ 42%] Building CXX object src/libs/browser/CMakeFiles/Browser.dir/webview.cpp.o
/nix/var/nix/builds/nix-60139-4043517989/source/src/libs/ui/qxtglobalshortcut/qxtglobalshortcut_mac.cpp:56:35: error: 'activateShortcut' is a protected member of 'QxtGlobalShortcutPrivate'
   56 |         QxtGlobalShortcutPrivate::activateShortcut(id.second, id.first);
      |                                   ^
/nix/var/nix/builds/nix-60139-4043517989/source/src/libs/ui/qxtglobalshortcut/qxtglobalshortcut_p.h:76:17: note: declared protected here
   76 |     static bool activateShortcut(quint32 nativeKey, quint32 nativeMods);
      |                 ^
1 error generated.
make[2]: *** [src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/build.make:114: src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/qxtglobalshortcut_mac.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:928: src/libs/ui/qxtglobalshortcut/CMakeFiles/QxtGlobalShortcut.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 42%] Built target Registry_autogen
[ 43%] Linking CXX static library libSidebar.a
[ 44%] Linking CXX static library libWidgets.a
[ 45%] Built target Sidebar
[ 46%] Built target Widgets
[ 48%] Linking CXX static library libBrowser.a
[ 49%] Built target Browser
make: *** [Makefile:111: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.4.0, macOS 26.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.3+1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/gay80fqbpm2wakbsyd4in44gx0cwx3h5-source`



### Notify maintainers

@peterhoeg 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ze/zeal/package.nix
+++ b/pkgs/by-name/ze/zeal/package.nix
@@ -11,6 +11,7 @@
   libpthread-stubs,
   libxcb-keysyms,
   qt6,
+  fetchpatch,
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "zeal";
@@ -22,6 +23,16 @@ stdenv.mkDerivation (finalAttrs: {
     tag = "v${finalAttrs.version}";
     hash = "sha256-FGg89bluN2IJJtkjwPa6dC83CBLdOr+LW5ArUKp4awk=";
   };
+
+  patches = [
+    # https://github.com/zealdocs/zeal/issues/1813
+    # Can likely remove with 0.9
+    (fetchpatch {
+      name = "fix-activateShortcut-protected.patch";
+      url = "https://github.com/zealdocs/zeal/commit/f3714111ecad65ddedde43fc7c4f8c5bd240ff64.patch";
+      hash = "sha256-DKTvanO14NRFhiHayJIWXWO7gQSRyjCQ1XFAiEN86XI=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake

